sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/execution_dispatcher.py

```diff
@@ -107,12 +108,27 @@ async def _create_pipelined_execution_context(self, query_execution_info):
                 raise CosmosHttpResponseError(StatusCodes.BAD_REQUEST,
                                   "Cross partition query only supports 'VALUE <AggregateFunc>' for aggregates")
 
-        execution_context_aggregator = multi_execution_aggregator._MultiExecutionContextAggregator(self._client,
+        # throw exception here for vector search query without limit filter
+        if query_execution_info.get_non_streaming_order_by():
+            if query_execution_info.get_top() is None and query_execution_info.get_limit() is None:
+                if os.environ.get('NO_ITEM_LIMIT_FOR_VECTOR_SEARCH', None) is None:
```

Comment:
Hey @simorenoh,
In Node SDK, we have done it little bit differently. If the query doesn't provide top/limit we do give client an option to provide buffer size from the request options:
[VectorSearchBufferSize](https://github.com/Azure/azure-sdk-for-js/blob/8a7635f10c713d4bc1545af4f8df5af26f563917/sdk/cosmosdb/cosmos/review/cosmos.api.md?plain=1#L1021)
[Size Deciding logic](https://github.com/Azure/azure-sdk-for-js/blob/8a7635f10c713d4bc1545af4f8df5af26f563917/sdk/cosmosdb/cosmos/src/queryExecutionContext/pipelinedQueryExecutionContext.ts#L296-L305)

Reply:
in both Java and Python we went ahead with returning an error if the user tries to use these queries without a limit